components/calendar/src/any_calendar.rs

```diff
@@ -942,46 +942,39 @@ impl AnyCalendarKind {
     /// Returns `None` if the calendar is unknown. If you prefer an error, use
     /// [`CalendarError::unknown_any_calendar_kind`].
     pub fn get_for_bcp47_value(x: &Value) -> Option<Self> {
-        let slice = x.as_tinystr_slice();
-
-        if slice.len() <= 2 {
-            if let Some(first) = slice.get(0) {
-                if let Some(second) = slice.get(1) {
-                    if first == "islamic" {
-                        match second.as_str() {
-                            "civil" => return Some(AnyCalendarKind::IslamicCivil),
-                            "tbla" => return Some(AnyCalendarKind::IslamicTabular),
-                            "umalqura" => return Some(AnyCalendarKind::IslamicUmmAlQura),
-                            _ => (),
-                        }
-                    }
-                } else {
-                    match first.as_str() {
-                        "buddhist" => return Some(AnyCalendarKind::Buddhist),
-                        "chinese" => return Some(AnyCalendarKind::Chinese),
-                        "coptic" => return Some(AnyCalendarKind::Coptic),
-                        "dangi" => return Some(AnyCalendarKind::Dangi),
-                        "ethioaa" => return Some(AnyCalendarKind::EthiopianAmeteAlem),
-                        "ethiopic" => return Some(AnyCalendarKind::Ethiopian),
-                        "gregory" => return Some(AnyCalendarKind::Gregorian),
-                        "hebrew" => return Some(AnyCalendarKind::Hebrew),
-                        "indian" => return Some(AnyCalendarKind::Indian),
-                        "islamic" => return Some(AnyCalendarKind::IslamicObservational),
-                        "islamicc" => return Some(AnyCalendarKind::IslamicCivil),
-                        "iso" => return Some(AnyCalendarKind::Iso),
-                        "japanese" => return Some(AnyCalendarKind::Japanese),
-                        "japanext" => return Some(AnyCalendarKind::JapaneseExtended),
-                        "persian" => return Some(AnyCalendarKind::Persian),
-                        "roc" => return Some(AnyCalendarKind::Roc),
-                        _ => (),
-                    }
-                }
+        match *x.as_tinystr_slice() {
```

Comment:
did clippy do this?

Reply:
impressive

Reply:
no I did this because I didn't like using a mix of `.first()` and `.get(1)`